lib/mysql.php

```diff
@@ -61,9 +61,15 @@ function mysql_connect(
                 return \Dshafik\MySQL::$connections[$hash]['conn'];
             }
 
+            if (strpos($hostname, ':') !== false) {
+                list($hostname, $port) = explode(':', $hostname, 2);
```

Comment:
See comment on line 64 for why this won't work correctly in all cases. I suggest capturing the index `$i` in the `strpos()` call above. Then
```
$hostandport = $hostname;
$hostname = substr($hostandport, 0, $i);
$port = (int)substr($hostandport, $i + 1);
```